app/models/package_manager/base.rb

```diff
@@ -132,42 +105,57 @@ def self.update_version(name, version)
         return
       end
 
-      dbproject = Project.find_by!(name: mapped_project[:name], platform: db_platform)
-      add_version(dbproject, one_version(dbproject.name, version))
+      db_project = Project.find_by!(name: mapped_project[:name], platform: db_platform)
+      add_version(db_project, one_version(db_project.name, version))
       # TODO handle deprecation here too
       save_dependencies(mapped_project) if self::HAS_DEPENDENCIES
-      finalize_dbproject(dbproject)
+      finalize_db_project(db_project)
     end
 
-    def self.add_version(dbproject, version_hash)
-      existing = dbproject.versions.find_or_initialize_by(number: version_hash[:number]) do |new_version|
-        new_version.update_attributes(version_hash)
+    def self.add_version(db_project, version_hash)
+      existing = db_project.versions.find_or_initialize_by(number: version_hash[:number]) do |new_version|
+        new_version.assign_attributes version_hash
       end
       existing.repository_sources = Set.new(existing.repository_sources).add(self::REPOSITORY_SOURCE_NAME).to_a if self::HAS_MULTIPLE_REPO_SOURCES
       existing.save
     end
 
-    def self.deprecate_versions(dbproject, version_hash)
-      case dbproject.platform.downcase
+    def self.deprecate_versions(db_project, version_hash)
+      case db_project.platform.downcase
       when "rubygems" # yanked gems will be omitted from project JSON versions
-        dbproject
+        db_project
           .versions
           .where.not(number: version_hash.pluck(:number))
           .update_all(status: "Removed")
       end
     end
 
-    def self.finalize_dbproject(dbproject)
-      dbproject.reload
-      dbproject.download_registry_users
-      dbproject.last_synced_at = Time.now
-      dbproject.save
-      dbproject
+    def self.finalize_db_project(db_project)
+      db_project.reload
+      db_project.download_registry_users
+      db_project.update(last_synced_at: Time.now)
+      db_project
     end
 
     def self.update(name, sync_versions: true)
       proj = project(name)
-      save(proj, sync_versions: sync_versions) if proj.present?
+      return unless project.present?
+
+      mapped_project = map_project(project)
+      return false unless mapped_project.present?
+
+      db_project = Project.find_or_initialize_by({ name: mapped_project[:name], platform: db_platform })
+      db_project.reformat_repository_url unless db_project.new_record?
+      db_project.update(mapped_project.except(:name, :releases, :versions, :version, :dependencies, :properties))
+
+      if self::HAS_VERSIONS && sync_versions
```

Comment:
Wonder if a later cleanup would be relying on all the stuff in update_version and just iterating through versions in the update method.